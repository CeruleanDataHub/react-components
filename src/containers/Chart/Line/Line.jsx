import React from "react";

import { Chart } from "../Chart";

export const Line = ({ title, xAxis, series }) => (
  <Chart title={title} xAxis={xAxis} series={series} type="line" />
);

Line.propTypes = {
  /** Title of the chart */
  title: Chart.propTypes.title,
  /** Values for X axis */
  xAxis: Chart.propTypes.xAxis,
  /** Data to be graphed */
  series: Chart.propTypes.series
};

Line.defaultProps = {
  title: Chart.defaultProps.title,
  xAxis: Chart.defaultProps.xAxis,
  series: Chart.defaultProps.series
};
