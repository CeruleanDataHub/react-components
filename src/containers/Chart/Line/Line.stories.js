import React from "react";

import { Line } from "./Line";

export default {
  title: "Line",
  component: Line
};

const xAxis = [
  {
    categories: [
      "monday",
      "tuesday",
      "wednesday",
      "thursday",
      "friday",
      "saturday",
      "sunday"
    ]
  }
];

const series = [
  { name: "Temperature", data: [99, 80, 82, 59, 90, 6, 99] },
  { name: "Humidity", data: [29, 39, 26, 53, 9, 46, 31] }
];

const legendDisabledOptions = { legend: { enabled: false } };

export const LineStory = () => (
  <Line title="Sauna histogram as line" xAxis={xAxis} series={series} />
);

export const LineLegendDisabledStory = () => (
  <Line
    title="Sauna temperature"
    xAxis={xAxis}
    series={[series[0]]}
    options={legendDisabledOptions}
  />
);
