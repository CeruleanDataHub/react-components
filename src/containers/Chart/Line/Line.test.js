import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import { Line } from "./Line";

describe("Line", () => {
  it("should render line chart", () => {
    const component = shallow(<Line />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it("should render chart with simple data", () => {
    const xAxis = [
      {
        categories: [
          "monday",
          "tuesday",
          "wednesday",
          "thursday",
          "friday",
          "saturday",
          "sunday"
        ]
      }
    ];

    const series = [
      { name: "temperature", data: [10, 15, 20, 25, 15, 20, 15] }
    ];

    const component = shallow(<Line xAxis={xAxis} series={series} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
