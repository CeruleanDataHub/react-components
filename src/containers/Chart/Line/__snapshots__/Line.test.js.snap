// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line should render chart with simple data 1`] = `
<Chart
  options={Object {}}
  series={
    Array [
      Object {
        "data": Array [
          10,
          15,
          20,
          25,
          15,
          20,
          15,
        ],
        "name": "temperature",
      },
    ]
  }
  title=""
  type="line"
  xAxis={
    Array [
      Object {
        "categories": Array [
          "monday",
          "tuesday",
          "wednesday",
          "thursday",
          "friday",
          "saturday",
          "sunday",
        ],
      },
    ]
  }
/>
`;

exports[`Line should render line chart 1`] = `
<Chart
  options={Object {}}
  series={Array []}
  title=""
  type="line"
  xAxis={null}
/>
`;
